BrainPortal/app/models/normal_user.rb

```diff
@@ -27,16 +27,32 @@ class NormalUser < User
 
   Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
 
+
   def available_tools  #:nodoc:
-    Tool.where( ["tools.user_id = ? OR tools.group_id IN (?)", self.id, self.group_ids])
+    available_group_ids    = (self.available_groups.pluck(:id) + self.group_ids).uniq
```

Comment:
I'm not sure about these changes (in this file and the next) about `available_XYZ()`. I think the original intent was to provide mehtods that return the availability of some models just based on their groupship, irrespective of if the model was associated with another model with its own restrictions (e.g. tasks on a bourreau, tools on a bourreau, tool_configs on a tool etc)

Reply:
I was thinking more about just enhancing these methods so that the model is available it if belongs to a public group... not all that much more.